src/Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -1,47 +1,140 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
 using System.CommandLine;
 using System.CommandLine.Invocation;
-using System.IO;
+using System.Text;
 using System.Threading.Tasks;
 using Microsoft.DotNet.Interactive;
 using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.CSharp;
 using Microsoft.DotNet.Interactive.Events;
-using Microsoft.DotNet.Interactive.Formatting;
 
 namespace Microsoft.Data.Analysis.Interactive
 {
     public class DataFrameKernelExtension : IKernelExtension
     {
-        public Task OnLoadAsync(IKernel kernel)
+
+        public string GetFriendlyName(Type type)
         {
-            //Formatter<DataFrame>.Register((tree, writer) =>
-            //{
-            //    writer.Write("");
-            //}, "text/html");
+            string friendlyName = type.Name;
+            if (type.IsArray)
+            {
+                // Not handled yet
+                return "DataFrameColumn";
+            }
+            if (type.IsGenericType)
+            {
+                int backTick = friendlyName.IndexOf('`');
+                if (backTick > 0)
+                {
+                    friendlyName = friendlyName.Remove(backTick);
+                }
+                friendlyName += "<";
+                Type[] typeParameters = type.GetGenericArguments();
+                for (int i = 0; i < typeParameters.Length; ++i)
+                {
+                    string typeParamName = GetFriendlyName(typeParameters[i]);
+                    friendlyName += (i == 0 ? typeParamName : "," + typeParamName);
+                }
+                friendlyName += ">";
+            }
+
+            return friendlyName;
+        }
+
+        public string GetTypedDataFrameWithProperties(DataFrame dataFrame, string resultTypeName, out StringBuilder prettyFormatter)
+        {
+            prettyFormatter = new StringBuilder();
+            StringBuilder stringBuilder = new StringBuilder();
+            string constructor = @$"
+public class {resultTypeName} : DataFrame 
+{{
+    public {resultTypeName}(DataFrame dataFrame)
+    {{
+        foreach (var column in dataFrame.Columns)
+        {{
+            Columns.Add(column);
+        }}
+    }}
+
+";
+            stringBuilder.Append(constructor);
+            prettyFormatter.Append(constructor);
 
+            foreach (var column in dataFrame.Columns)
+            {
+                string columnName = column.Name.Replace(" ", string.Empty);
+                Type dataType = column.DataType;
+                string typeName = GetFriendlyName(column.GetType());
+                stringBuilder.Append($@"
+    public {typeName} {columnName} 
+    {{
+        get
+        {{
+            int columnIndex = Columns.IndexOf(""{columnName}"");
```

Comment:
Why isn't this just `return Columns[""{columnName}""];`?

Reply:
Because `Columns[columnName]` doesn't exist on the `DataFrame` side in 0.2.0.

Reply:
Then why not `return this[""{columnName}""];`?

Reply:
Because `return this["columnName"]` will go away in the next preview in favor of `Column[columnName]`.